docker/object_store/docker-compose.yml

```diff
@@ -0,0 +1,116 @@
+version: '3.8'
+services:
+  minio1:
+    image: minio/minio:latest
+    hostname: minio1
+    volumes:
+      - minio1-data1:/export1
+      - minio1-data2:/export2
+      #- minio1-data3:/export3
+      #- minio1-data4:/export4
+      #- minio1-data5:/export5
+      #- minio1-data6:/export6
+    #ports:
+    #     - "9001:9000"
+    expose:
+      - 9000
+    networks:
+      - mpi-net
+      - requests-net
+    deploy:
+      restart_policy:
+        delay: 10s
+        max_attempts: 10
+        window: 60s
+      placement:
+        constraints:
+          - node.labels.minio1==true
```

Comment:
On first glance, my thought for this was we need to either automate adding these labels or at least automate a check for them prior to this stage of starting the stack.  Thinking more, that is less than ideal because of the ambiguity of how any arbitrary swarm might be set up, and exactly where nodes might need to be placed. But even so, it seems to me that if we are defining each instance with its own service - effectively fixing the size of the minio cluster - it may still be best to implement direct control over the node setup/placement. 

That said, I'm wondering if we shouldn't try a different approach. Instead of multiple services, just have one service definition, but have it set up with replicas. This would also seem to make the minio cluster size easily configurable per environment. Swarm node labels could still be used locally for controlling placement, and it would be more sensible to require this to be set up manually if it enabled greater flexibility and control. 

It's possible you already considered all this and the constrains of swarm or minio prevent it somehow. Otherwise, might be worth talking through in a bit more detail.

Reply:
The problem is with the underlying storage.  If you had a distributed file system where some other service guaranteed replication of data across the hosts, then you could do this.  Docker swarm does not replicate volumes (they are local to each node) so you have to pin the service to the host its volume is associated with.